scripts/pin_oldest_deps.py

```diff
@@ -15,16 +15,30 @@ def main(args: list[str]) -> int | str | None:
     if args:
         return f"Unsupported arguments: {args!r}"
     pyproject = tomlkit.loads(pyproject_path.read_text())
+
     poetry_deps = pyproject["tool"]["poetry"]["dependencies"]  # type: ignore[index]
     assert isinstance(poetry_deps, AbstractTable)
+    _pin_oldest_for_deps_list(poetry_deps)
+
+    dev_deps = pyproject["tool"]["poetry"]["group"]["dev"]["dependencies"]  # type: ignore[index]
+    assert isinstance(dev_deps, AbstractTable)
+    _remove_duplicate_dev_deps(poetry_deps, dev_deps)
+
+    pyproject_path.write_text(tomlkit.dumps(pyproject))
+    print("Updated pyproject.toml with pinned dependencies.")
+    return None
 
-    for dep, value in poetry_deps.items():
+
+def _pin_oldest_for_deps_list(deps_list: AbstractTable) -> None:
+    assert isinstance(deps_list, AbstractTable)
+
+    for dep, value in deps_list.items():
```

Comment:
The caller already does this.
```suggestion
def _pin_oldest_for_deps_list(deps_list: AbstractTable) -> None:
    for dep, value in deps_list.items():
```

Reply:
No, that was just a copy/paste mistake from when I was refactoring into a separate method. That check isn't needed anymore. Thanks for catching that.